### DISCUSSION TITLE
How many times we can associate a Work Item to be appeared on generated Release Note

### DISCUSSION THREAD

**Discussion by khopithan** [rfennell/AzurePipelines#1359] [Q&A] [answered]
Hi @rfennell,

When I was doing some testing with Pipelines and Work Items, I noticed one thing.

If I associate a particular Work Item with the commit for multiple times, I can see the associate commits details and pipeline run details in the Sprint board (on Work Item)

But When I check the Release note I couldn't see the associated Work Items related details. So, my question is....

**How many times we can associate a particular Work Item to be appeared on generated Release Note?**

Regards,
Khopi

**Comment by rfennell** [comment]
You have not stated if you are using multistage YAML or Classic Releases - I am from the context of question assuming Classic Releases.

My task in Classic Releases does the following

1. Finds the last successful release
2. For each build artifact
    1. Find the ID of the build in the current release and the last successful release
    2. Gets the commit & WI between these builds 
    3. Removes duplicates
    4. Add the details to the of the build inc WI and Commits to the `builds` array
    5. Adds the WI & Commits to the global list
3. Removes duplicates WI and Commits
4. Exposes the lobal list as the `workitems` and `commits` arrays

There is nothing here that stops work items being shown if used multiple times. I am just making Azure DevOps API calls and meging the result sets

All I can suggest is doing [local  debugging](https://github.com/rfennell/AzurePipelines/blob/main/Extensions/XplatGenerateReleaseNotes/V3/testconsole/readme.md)

**Reply by khopithan** [reply]
Hi @rfennell,

Thank you for the reply. I have multi stage YAML pipeline.

Regards,
Khopi

**Comment by rfennell** [comment, marked as answer]
In that case it is simpler, it just gets uses the API call to get the WI and CS between builds. [The logic is all around ln 1410 ](https://github.com/rfennell/AzurePipelines/blob/main/Extensions/XplatGenerateReleaseNotes/V3/ReleaseNotesFunctions.ts)

It then tries to get detail of associate artifacts, and passes this to the handlebars processor. So again nothing to filter items, I just make standard API calls

**Reply by khopithan** [reply]
Thank you...I'll check this.

Khopi
